src/internal/packager/helm/chart.go

```diff
@@ -118,6 +121,33 @@ func (h *Helm) InstallOrUpgradeChart(ctx context.Context) (types.ConnectStrings,
 		return nil, "", installErr
 	}
 
+	resourceList, err := h.actionConfig.KubeClient.Build(bytes.NewBufferString(release.Manifest), true)
+	if err != nil {
+		return nil, "", fmt.Errorf("unable to build the resource list: %w", err)
+	}
+
+	healthChecks := []v1alpha1.NamespacedObjectKindReference{}
+	for _, resource := range resourceList {
+		apiVersion, kind := resource.Object.GetObjectKind().GroupVersionKind().ToAPIVersionAndKind()
+		healthChecks = append(healthChecks, v1alpha1.NamespacedObjectKindReference{
+			APIVersion: apiVersion,
+			Kind:       kind,
+			Name:       resource.Name,
+			Namespace:  resource.Namespace,
+		})
+	}
+	if !h.chart.NoWait {
+		// This re-uses the timeout from helm. This will increase the total amount of time a timeout can take
+		// However it is unlikely this step will take long
+		healthChecksCtx, cancel := context.WithTimeout(ctx, h.timeout)
```

Comment:
Want to get the teams thoughts here. This way is much simpler, but may result in unexpected behavior. 

If we want to use the same context as the successful helm install, I believe we would have to use `retry.DoWithData` and make a custom object to send back both the context and the cancel function